File: app/src/main/java/tw/y_studio/ptt/Adapter/GeneralFragmentStatePagerAdapter.java
package tw.y_studio.ptt.Adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import java.util.ArrayList;

public class GeneralFragmentStatePagerAdapter extends FragmentStateAdapter {
    private ArrayList<Fragment> fragmentArrayList;

    public GeneralFragmentStatePagerAdapter(
            @NonNull FragmentActivity fm, @NonNull ArrayList<Fragment> fragmentArrayList) {
        super(fm);
        this.fragmentArrayList = fragmentArrayList;
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return fragmentArrayList.get(position);
    }

    @Override
    public int getItemCount() {
        return fragmentArrayList.size();
    }
}

File: app/src/main/java/tw/y_studio/ptt/ui/adapter/GeneralFragmentStatePagerAdapter.kt
package tw.y_studio.ptt.ui.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import java.util.*

class GeneralFragmentStatePagerAdapter(
    fm: FragmentActivity,
    private val fragmentArrayList: ArrayList<Fragment>
) : FragmentStateAdapter(fm) {
    override fun createFragment(position: Int): Fragment {
        return fragmentArrayList[position]
    }

    override fun getItemCount(): Int {
        return fragmentArrayList.size
    }
}
